dataactbroker/permissions.py

```diff
@@ -34,8 +34,9 @@ def decorated_function(*args, **kwargs):
                 elif LoginSession.isLogin(session):
                     user = sess.query(User).filter(User.user_id == session["name"]).one()
                     valid_user = True
-                    for permission in permission_list :
-                        if not has_permission(user, permission):
+                    for permission in permission_list:
+                        if permission in PERMISSION_TYPE_DICT and \
+                                not user.permission_type_id == PERMISSION_TYPE_DICT[permission]:
```

Comment:
I've added the louder exception handling and have removed the use of a list. You're right, we're not longer using a list for checking permissions anywhere (double checked since we're using this for checking email token and password token, which will be removed soon)